### DISCUSSION TITLE
"unsaved" work in Interactive Window is lost on reload

### DISCUSSION THREAD

**Discussion by amunger** [microsoft/vscode-jupyter#9583] [New Ideas]
(edited for discussion)

We don't currently persist anything in an Interactive Window between VS Code sessions, but is that something that users expect? Hot exit and auto-save features are pretty pervasive through the product (even a python terminal will come back up in the same state after a reload), so should the IW follow suit, or does it stand out as ephemeral?

Taken from #6460
What is the meaning of hot exit, and why does it exist?
- Is it just about persisting the history in the interactive window, or is it also about having a persistent kernel connection?
- One perspective: if I reload VS Code, I don't want to lose my interactive window session--that means the mapping between the Python file and interactive window should still be there, and the kernel connection should still be live and I should not lose my kernel state

**Comment by DonJayamanne** [comment]
@amunger lost all of the work that was in IW.

**Comment by Ark-kun** [comment]
I had code working in an interactive session. There was a lot of state and several task queues. At some point my VSCode window got very slow and stuck, so I did Reloaded Window. My kernel got killed and all my state was lost. It was really hard to recover from this data loss. I wish there was a way to prevent VSCode from killing running kernels.
